helix-term/src/ui/editor.rs

```diff
@@ -770,6 +770,72 @@ impl Component for EditorView {
 
                 EventResult::Consumed(callback)
             }
+            Event::Mouse(MouseEvent {
+                kind: MouseEventKind::Down(MouseButton::Left),
+                row,
+                column,
+                modifiers,
+                ..
+            }) => {
+                let editor = &mut cx.editor;
+
+                let result = editor.tree.views().find_map(|(view, _focus)| {
+                    if !view.verify_screen_coords(row as usize, column as usize) {
```

Comment:
Call `verify_screen_coords` inside `pos_at_screen_coords` and return an `Option` from pos_at_screen_coords. You can actually inline it completely and remove the function. Then all you need to do here is 

```rust
find_map(|(view, _focus)| {
//...;
view.pos_at_screen_coords(...).map(|coords| (coords, view.id))
})
```

Reply:
We've decided (cannot reference conversation, will try to find later) that it is not necessary to check inside it as we already validate view while searching for it.

Reply:
Although if they're united it can be better.